### DISCUSSION TITLE
The sensor "state" does not update in Home Assistant

### DISCUSSION THREAD

**Discussion by r2-4b** [PricelessToolkit/MailBoxGuard#58] [Q&A] [answered]
Everything works so far. I tested everything with the "OLED" version.
Afterward, I switched to MQTT for Home Assistant (HASS). The good thing is that it partially works.
I only adjusted the parameters specified by PricelessToolkit in the code, nothing else.
There is still a bug or an error I am making.

As soon as I trigger the motion sensor, the RSSI and battery status are updated immediately in HASS, but not the state.

When I click on MQTT in HASS, the sensor information appears. In the subscribed topics, I see the last 10 payloads and when they were triggered. That all seems correct, but the "state" sensor does not update and always remains on "Mail" or any other status I set it to. Even after triggering, it does not revert to the correct state.

Attached is the Diagnostic File from HASS:
[MQTT_Diagnose_Postsensor_from HASS.txt](https://github.com/user-attachments/files/17781982/MQTT_Diagnose_Postsensor_from.HASS.txt)

![image](https://github.com/user-attachments/assets/001e166f-6068-423d-9098-b1dfd0fd61a4)

![image](https://github.com/user-attachments/assets/3d47ec63-d2d8-4b24-aba9-9795d6900057)

Here i have triggered it  a few times and it stays "empty"
![image](https://github.com/user-attachments/assets/be8e22f6-4040-4865-845e-585f3f7aa04a)

RSSI working great, refreshes at every triggerig:
![image](https://github.com/user-attachments/assets/5896c84b-6e0b-42cd-90e3-a100f61bb01e)

What could be the cause of this? Because as it stands, I cannot use it for HASS yet :-(
I would really appreciate some help.

**Comment by PricelessToolkit** [comment]
Hi, To reset the "Mail" state, don't just set it from HA. Use MQTT publish like "empty" to the topic.

![mqtt_empty](https://github.com/user-attachments/assets/a64082ad-3126-4867-ad8e-9791b564ef79)

**Comment by r2-4b** [comment]
Hi,No, unfortunately, you did not understand the problem correctly. The status can be set to "Empty" without any issues. However, once the motion sensor is triggered, the status no longer changes to "Mail."

**Reply by PricelessToolkit** [reply, marked as answer]
1. Delete the sensor from the "HA" MQTT sensor list.
2. Trigger Sensor.
3. check if the status is "mail"
4. Publish "empty" to topic. "homeassistant/sensor/mbox/state" and don't just set the status to "empty"

If you change the status from HA even once, after that it will not work.
![mqtt_empty](https://github.com/user-attachments/assets/270e0be8-9c58-4a56-bbe1-720830c46a2c)


PS: You used this code, right?  https://github.com/PricelessToolkit/MailBoxGuard/tree/main/Code/MQTT_JSON

**Reply by r2-4b** [reply]
Hi,

now it works awesome and perfect as pretended. You are my Hero! The Range ist brutal! 
Yes i used 1:1 completely your instructions. only the Programmer was a jtag2updi wirth an Arduino Nano 👍 

I've deleted it like you said: 
![image](https://github.com/user-attachments/assets/813514c4-0acb-4df7-9c7b-00e33bba3488)

and after that it disappears and i restarted HA: 
![image](https://github.com/user-attachments/assets/3efa3c7f-7937-4f86-beb8-63d0dc3eca18)

Then i've tested it with MQTT Publish and Abo:
![image](https://github.com/user-attachments/assets/f63b4bea-8b31-4a20-add7-13548eb2d77e)
 
![image](https://github.com/user-attachments/assets/9f2e3996-0ea6-4492-985a-8833d531c02d)

It works now: 
![image](https://github.com/user-attachments/assets/111d8fef-f76c-4785-bfb9-713c3f4ef95a)

Thank you very much :-) 👌❤❤❤❤

**Reply by PricelessToolkit** [reply]
I'm glad everything worked out.
